src/systems/battle_hunger.rs

```diff
@@ -0,0 +1,56 @@
+use crate::*;
+
+/// Slows and damages the closest enemy unit for 5 seconds. Caster gains movement speed.
+/// TODO: End duration if target kills another unit
+pub fn battle_hunger_system(
+    positions: &Components<Point>,
+    teams: &Components<Team>,
+    entities: &Entities,
+    effector_defs: &EffectorDefinitions<Stats, Effectors>,
+    effectors: &mut Components<EffectorSet<Effectors>>,
+    events: &mut Vec<SkillTriggerEvent<Skills>>,
+    game_events: &mut Vec<GameEvent>,
+) -> SystemResult {
+    for ev in events.iter() {
+        for (e, effector) in join!(&entities && &effectors) {
+            let e = e.unwrap();
+            if let Some(effectors) = effector {
+                for effector in &effectors.effectors {
+                    if effector.effector_key == Effectors::Enraged {
+                        game_events.push(GameEvent::DamageEntity(e, 1.0));
```

Comment:
I'm thinking of adding a new type of effectors that will directly modify the base stat value, allowing for effectors that do damage each tick. That should be coming in the next few days in game_features and minigene.
There's a few design issues I still have to solve. For example, how we can use those types of effectors while still using properly the DamageEntity event to update the total damage value.

For now we'll keep this solution here.